warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java

```diff
@@ -2259,4 +2270,214 @@ public Equality getEquality() {
         }
         return equality;
     }
+
+    private static final Joiner COMMA_JOINER = Joiner.on(",").skipNulls();
+
+    // Map of class strings to DefaultOptions implementations.
```

Comment:
Overall, I would like to see more complete documentation here and for the new methods below that leverage javadoc and go into more complete detail about the contract defined here for specifying default options, as well as caveats for overriding them.